### ISSUE TITLE
Q about the importance score

### ISSUE THREAD

**Issue by digbangbang** [ROIM1998/APT#4]
Hi, I have some questions about the importance score in ur paper. Actuallay the importance score of one parameters or block in some other literatures is the weight value product the weight gradient, which is derived from the loss changed when set the parameter to zero and using Taylor Expand. So why using the following score is valid?

![image](https://github.com/user-attachments/assets/625af9b0-a7a9-4a0b-80e4-2b306b82fd40)

I know the gradient of weight is not available when in PEFT, and just want to know whether the alternative method is valid.

Thanks if u reply soon.

**Comment by ROIM1998** [comment]
Hello! Thanks for your interest in our work. Sorry I was busy with some working stuffs last Friday and didn't get your message replied.

We have introduced the deduction process of the salience scoring function here in Appendix B, where we notice that the block-wise salience scoring function of the activation is actually the summation of the salience score of the tuning LoRA weights and frozen backbone weights. We also add the batch-wise approximation here to compute the salience score more efficiently. If you have any further questions, please let me know!

**Comment by digbangbang** [comment]
Thanks, I've realize the decution. Some new questions for the experiments.

![image](https://github.com/user-attachments/assets/6d626c42-7aa3-4cc0-9b5a-13da7174328f)

In the experiment, I observed that sometimes the training time of APT is less than LoRA, and sometimes it is more than LoRA. From my personal understanding, I feel that because APT can prune parameters, its training time will be reduced accordingly, but why does it take more training time than LoRA? In your experiment here, do you set all the training steps to be the same, and then calculate the training time?
